LayoutTests/platform/mac/TestExpectations

```diff
@@ -2480,3 +2480,4 @@ imported/w3c/web-platform-tests/html/user-activation/activation-trigger-pointere
 
 webkit.org/b/242804 media/track/track-forced-subtitles-in-band.html [ Pass Failure ]
 
+webkit.org/b/251051 [ BigSur+ Debug ] storage/indexeddb/modern/deleteindex-4-private.html [ Crash ]
```

Comment:
`BigSur+` wasn't necessary here. Big Sur is the oldest OS supported, so limiting to that or newer is a no-op.